Sdp mediadrm support

petegriffin opened the pull request:
Adds support to the mediadrm plugin for decrypting into a SDP allocated buffer. Relies on a corresponding pull request to the ClearKey TA

petegriffin committed:
clearkey: propogate secure flag to lower layers.

Also call TEE_copy_secure_memory() function
if running in secure mode. This allows unencrypted
data to be copied into a secure buffer by the TEE.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

petegriffin committed:
AesCtrDecryptorUnittest.cpp: Add offset argument

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

petegriffin committed:
Support both Clear key plugin UUIDs.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

andrey-konovalov reviewed (approved):
Looks good to me

petegriffin committed:
AesCtrDecryptor: Move TEE_crypto_init() call into AesCtrDecryptor object.

TEE_crypto_init() allocates memory for IV and key currently.
It is linked to the lifetime of the AES decryptor object
so makes no sense to be called from SessionLibrary object.

Also this change allows the AesCtrDecryptorUnitTests to be run
without any modification making maintaiing the OPTEE ClearKey
and rebasing to new versions of AOSP easier.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

ajutras-linaro commented on AesCtrDecryptor.cpp:
```diff
@@ -61,8 +62,13 @@ android::status_t AesCtrDecryptor::decrypt(const android::Vector<uint8_t>& key,
         if (subSample.mNumBytesOfClearData > 0) {
+	  if (secure)
+	    TEE_copy_secure_memory(destination, source,
+				   subSample.mNumBytesOfClearData, offset);
```
Is this API defined when USE_AES_TA is not defined?

petegriffin replied:
No it isn't good point. Will fix.

ajutras-linaro reviewed (changes requested):
See inline comment.

petegriffin committed:
README.md: add info about make rules for plugin & ta

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

petegriffin committed:
SessionLibrary: Add plugin name to debug output

Because aosp default sw clearkey plugin may also be installed,
it is useful to have the plugin name in the logs to ensure
you are using the correct plugin.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

petegriffin committed:
TEE_copy_secure_memory: swap src/dst params for consistency

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>
